src/main/java/com/idrsolutions/microservice/BaseServletContextListener.java

```diff
@@ -241,4 +243,20 @@ private void validateFileDeletionServiceFrequency(final Properties properties) {
             LOG.log(Level.WARNING, message);
         }
     }
+
+    private static void validateMaxConversionDuration(final Properties properties) {
+        final String maxDuration = properties.getProperty(KEY_PROPERTY_MAX_CONVERSION_DURATION);
+        if (maxDuration == null || maxDuration.isEmpty() || "0".equals(maxDuration) || !(maxDuration.matches("\\d+") || "Infinity".equals(maxDuration))) {
+            properties.setProperty(KEY_PROPERTY_MAX_CONVERSION_DURATION, String.valueOf(Long.MAX_VALUE));
+            final String message = String.format("Properties value for \"maxConversionDuration\" was set to " +
+                    "\"%s\" but should be a positive long or Infinity. Using a value of Infinity.", maxDuration);
+            LOG.log(Level.WARNING, message);
+            return;
+        }
+        if ("Infinity".equals(maxDuration)) {
```

Comment:
We can move this if statement into the one above.
I suggest after the property is set you use this if statement to either log a message for infinity or a message or an invalid value.

Reply:
I'm not sure I follow. The "Infinity".equals(maxDuration) part in the first if statement is negated so moving the second if statement inside it would mean its condition is never true.
We could make line 256 an else if, removing the need for the return on 254.

Reply:
Apologies, I misread. Happy with the changes